cloudinit/cmd/clean.py

```diff
@@ -132,14 +134,29 @@ def remove_artifacts(init, remove_logs, remove_seed=False, remove_config=None):
     ):
         for conf in GEN_SSH_CONFIG_FILES:
             del_file(conf)
-    if remove_config and set(remove_config).intersection(
-        ["all", "datasource"]
-    ):
 
-        init.fetch().clean()
+    clean_datasource = remove_config and set(remove_config).intersection(
+        ["all", "datasource"]
+    )
 
     if not os.path.isdir(init.paths.cloud_dir):
+        log_util.multi_log(
+            "Artifacts already cleaned!",
```

Comment:
Nit: No need for exclamations, I think:

```suggestion
            "Artifacts already cleaned.",
```